src/tex.js

```diff
@@ -203,8 +231,26 @@ MathJax.Hub.Register.StartupHook('TeX Jax Ready', function () {
       return token;
     },
     arabicNumber: (function () {
-      var englishNumbersRegExp = /[0-9]/g;
-      var numbersMap = MathJax.Hub.config.Arabic.numbersMap;
+      // check for MathJax options
+      useHindiNumeral = MathJax.Hub.config.useHindiNumeral
+      useArabicDecimalSeparator = MathJax.Hub.config.useArabicDecimalSeparator
+      
+      // Default options
+      if (useHindiNumeral == null) useHindiNumeral = true
+      if (useArabicDecimalSeparator == null) useArabicDecimalSeparator = true
+
+      arConfig = MathJax.Hub.config.Arabic
+      if(useArabicDecimalSeparator){
+        var englishNumbersRegExp = /[0-9]\.?/g;
+        var numbersMap = {};
+        for (var nma in arConfig.numbersMap) { numbersMap[nma] = arConfig.numbersMap[nma]; }
+        for (var nma in arConfig.arabicDecimalSeparator) { numbersMap[nma] = arConfig.arabicDecimalSeparator[nma]; }
+      } else{
+        var englishNumbersRegExp = /[0-9]/g;
+        if(useHindiNumeral) var numbersMap = arConfig.numbersMap;
+        else var numbersMap = arConfig.arabicNumbersMap
+      }
```

Comment:
All the section above needs refactoring:

 - The for looop isn't protected with `hasOwnProperty`
 - Avoid for-loops and use RegExp replace functions like `var mapped = text.replace(englishNumbersRegExp, replaceNumber);`